### DISCUSSION TITLE
Does Valheim expose these functions to you, or did you create them?

### DISCUSSION THREAD

**Discussion by ghost** [Digitalroot-Valheim/Digitalroot.Valheim.MaxDungeonRooms#7] [General]
```c#
for (int i = 0; i < maxRooms; i++)
{
  TryToPlaceOneRoom();
  if (HasAllRequiredRooms() && placedRooms.Count > minRooms)
  {
break;
  }
}
```

Are these functions you created, or does Valheim expose them to you?  Without knowing anything about the code in these functions, I'm seeing flaws in your design (unnecessary logic).

**Comment by Digitalroot** [comment]
Hi @angelbearuk, 

That code is my paraphrasing code written by Iron Gate. For the sake of discussion, would you elaborate on the flaws.
